app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt

```diff
@@ -1615,6 +1620,66 @@ class ChatController(args: Bundle) :
                         }
                         true
                     }
+                    R.id.action_reply_privately -> {
+                        val apiVersion = ApiUtils.getConversationApiVersion(conversationUser, intArrayOf(ApiUtils.APIv4, 1))
+                        val retrofitBucket = ApiUtils.getRetrofitBucketForCreateRoom(
+                            apiVersion,
+                            conversationUser?.baseUrl,
+                            "1",
+                            message?.user?.id?.substring(6),
+                            null
+                        )
+                        ncApi!!.createRoom(
+                            credentials,
+                            retrofitBucket.getUrl(), retrofitBucket.getQueryMap()
+                        )
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribe(object : Observer<RoomOverall> {
+                                override fun onSubscribe(d: Disposable) {}
+                                override fun onNext(roomOverall: RoomOverall) {
+                                    val bundle = Bundle()
+                                    bundle.putParcelable(KEY_USER_ENTITY, conversationUser)
+                                    bundle.putString(KEY_ROOM_TOKEN, roomOverall.getOcs().getData().getToken())
+                                    bundle.putString(KEY_ROOM_ID, roomOverall.getOcs().getData().getRoomId())
+
+                                    // FIXME once APIv2 or later is used only, the createRoom already returns all the data
+                                    ncApi!!.getRoom(
+                                        credentials,
+                                        ApiUtils.getUrlForRoom(
+                                            apiVersion, conversationUser?.baseUrl,
+                                            roomOverall.getOcs().getData().getToken()
+                                        )
+                                    )
+                                        .subscribeOn(Schedulers.io())
+                                        .observeOn(AndroidSchedulers.mainThread())
+                                        .subscribe(object : Observer<RoomOverall> {
+                                            override fun onSubscribe(d: Disposable) {}
+                                            override fun onNext(roomOverall: RoomOverall) {
+                                                bundle.putParcelable(
+                                                    KEY_ACTIVE_CONVERSATION,
+                                                    Parcels.wrap(roomOverall.getOcs().getData())
+                                                )
+                                                remapChatController(
+                                                    router, conversationUser!!.id,
```

Comment:
better to use one from the 3 other suggestions:
![grafik](https://user-images.githubusercontent.com/2932790/118240596-7a512180-b49b-11eb-8342-5d69145aeb63.png)

only use !! when 100% sure it can't be null

Reply:
It can not be null as only logged in users should see the option and guests can't end up in here as also the createRoom would fail for them when they would call it.

But I fixed an issue that didn't hide the option when "Join with a link"